### ISSUE TITLE
Unable to set keyword_length > 2

### ISSUE THREAD

**Issue by pfries** [uga-rosa/cmp-dictionary#15]
Hi @uga-rosa, thanks for the great work on this plugin. I have it working fine when I have `keyword_length <= 2` but anything above that disables Dictionary source completion.

```lua
    sources = {
        { name = "dictionary", keyword_length = 5 }
    }
```

**Comment by uga-rosa** [comment]
What happens if you set `exact` to more than `keyword_length`?

**Comment by pfries** [comment]
Setting `keyword_length =4` and `exact = 5` produces an exception:

```
Error detected while processing TextChangedI Autocommands for "*":                                                                                                                                                   
E5108: Error executing lua ...packer/start/cmp-dictionary/lua/cmp_dictionary/merge.lua:160: attempt to get length of field 'array' (a nil value)                                                                     
stack traceback:                                                                                                                                                                                                     
        ...packer/start/cmp-dictionary/lua/cmp_dictionary/merge.lua:160: in function 'merge'                                                                                                                         
        .../packer/start/cmp-dictionary/lua/cmp_dictionary/init.lua:38: in function 'get_candidate'                                                                                                                  
        .../packer/start/cmp-dictionary/lua/cmp_dictionary/init.lua:53: in function 'complete'                                                                                                                       
        .../nvim/site/pack/packer/start/nvim-cmp/lua/cmp/source.lua:290: in function 'complete'                                                                                                                      
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/core.lua:250: in function 'complete'                                                                                                                      
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/core.lua:163: in function 'callback'                                                                                                                      
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/core.lua:213: in function 'autoindent'                                                                                                                    
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/core.lua:155: in function 'on_change'                                                                                                                     
        ...re/nvim/site/pack/packer/start/nvim-cmp/lua/cmp/init.lua:301: in function 'callback'                                                                                                                      
        ...ite/pack/packer/start/nvim-cmp/lua/cmp/utils/autocmd.lua:31: in function 'emit'                                                                                                                           
        [string ":lua"]:1: in main chunk
```

**Comment by pfries** [comment]
Still not seeing the completion source with `keyword_length = 4` and `exact = 5` or `exact = 4`.

**Comment by uga-rosa** [comment]
Hmm. It doesn't reproduce. Please provide detailed settings and steps to reproduce.
![Animation](https://user-images.githubusercontent.com/82267684/147865982-74bbfa3e-448c-4591-822c-9f002a348228.gif)

**Comment by pfries** [comment]
Still seeing it with a minimal `cmp` config, all plugins except `cmp` and `cmp-dictionary` removed, and no custom configuration.

```sh
nvim --version
NVIM v0.7.0-dev+2261-g9804a2870
```

```lua
require("cmp").setup({
    sources = {
        {
            name = "dictionary",
            keyword_length = 4
        },
    },
})

require("cmp_dictionary").setup({
    dic = {
        ["*"] = { "/usr/share/dict/words" },
    },
})

```

[![asciicast](https://asciinema.org/a/BFTHbJAxkA64mA9YrqRz3zHtq.svg)](https://asciinema.org/a/BFTHbJAxkA64mA9YrqRz3zHtq)

**Comment by uga-rosa** [comment]
Can you try to reproduce it in the release version?

**Comment by pfries** [comment]
Same behavior with v0.6.1

**Comment by uga-rosa** [comment]
Even v0.6.1 didn't reproduce it for me... I wonder what could be the cause?

**Comment by pfries** [comment]
Not sure but it sounds like it's something on my end I'll need to troubleshoot. Thanks for looking into it.

**Comment by uga-rosa** [comment]
Did you solve the problem? I'd like to close this issue, since there are no similar reports and the problem probably depends on your environment.

**Comment by pfries** [comment]
FYI this was fixed by 9ae00514826bd26b84d86d67ed999cb4155c33e4

Thanks!
